src/vellum/workflows/state/context.py

```diff
@@ -145,7 +147,32 @@ def resolve_workflow_deployment(self, deployment_name: str, release_tag: str) ->
         Returns:
             BaseWorkflow instance if found, None otherwise
         """
-        return None
+        if not self.generated_files:
+            return None
+
+        expected_hash = str(uuid4_from_hash(f"{deployment_name}|{release_tag}")).replace("-", "_")
+        expected_prefix = f"vellum_workflow_deployment_{expected_hash}"
+
+        workflow_file_key = f"{expected_prefix}/workflow.py"
+        if workflow_file_key not in self.generated_files:
+            return None
+
+        from vellum_ee.workflows.server.virtual_file_loader import VirtualFileFinder
+
+        finder = VirtualFileFinder(self.generated_files, expected_prefix)
+        sys.meta_path.append(finder)
+
+        try:
+            from vellum.workflows.workflows.base import BaseWorkflow
+
+            WorkflowClass = BaseWorkflow.load_from_module(expected_prefix)
+            workflow_instance = WorkflowClass(context=self)
```

Comment:
we want:
- `context=WorkflowContext.create_from(self)`
- `parent_state=state`

This means we should pass in `start: BaseState` into this resolve method